pkg/ipcache/cidr.go

```diff
@@ -18,78 +18,6 @@ import (
 	"github.com/cilium/cilium/pkg/source"
 )
 
-// AllocateCIDRs attempts to allocate identities for a list of CIDRs. If any
-// allocation fails, all allocations are rolled back and the error is returned.
-// When an identity is freshly allocated for a CIDR, it is added to the
-// ipcache if 'newlyAllocatedIdentities' is 'nil', otherwise the newly allocated
-// identities are placed in 'newlyAllocatedIdentities' and it is the caller's
-// responsibility to upsert them into ipcache by calling upsertGeneratedIdentities().
-//
-// Upon success, the caller must also arrange for the resulting identities to
-// be released via a subsequent call to ReleaseCIDRIdentitiesByCIDR().
-//
-// Deprecated: Prefer UpsertLabels() instead.
-func (ipc *IPCache) AllocateCIDRs(
-	prefixes []netip.Prefix, newlyAllocatedIdentities map[netip.Prefix]*identity.Identity,
-) ([]*identity.Identity, error) {
-	// maintain list of used identities to undo on error
-	usedIdentities := make([]*identity.Identity, 0, len(prefixes))
-
-	// Maintain list of newly allocated identities to update ipcache,
-	// but upsert them to ipcache only if no map was given by the caller.
-	upsert := false
-	if newlyAllocatedIdentities == nil {
-		upsert = true
-		newlyAllocatedIdentities = map[netip.Prefix]*identity.Identity{}
-	}
-
-	allocateCtx, cancel := context.WithTimeout(context.Background(), option.Config.IPAllocationTimeout)
```

Comment:
It looks like `option.Config.IPAllocationTimeout` is now unused, so I guess we could deprecate the corresponding `ip-allocation-timeout` option for v1.16 and remove it for v1.17.

Reply:
Good catch! I decided to just remove the flag. It would be a no-op in v1.16, I don't think we need to keep a no-op flag around.

Reply:
AFAIR we usually deprecate user-visible config flags for one release to give users a chance to adjust their config. But I can't find a section in the docs specifying this, only the release notes listing deprecated/removed flags, i.e. https://docs.cilium.io/en/stable/operations/upgrade/#deprecated-options

But given that the `ip-allocation-timeout` option is not used in the Helm chart it's probably OK to just remove it.